### PR TITLE
Fix maintenance mode PR

### PR DESCRIPTION
**Pull request by fabclj**
This PR includes again the changes already done in #351 and removed because conflict solving issues.

**How to test:**
put this as settings:
```
settings: {
    awaitEndpointConfig: true
}
```

Just set maintenance mode or Business hour on target endpoint and check if it appears as expected

### CODE REVIEW

File: src/webchat-ui/components/WebchatUI.tsx [hunk 1]
```diff
@@ -755,7 +781,12 @@ export class WebchatUI extends React.PureComponent<
 			);
 
 		const handleOnClose = () => {
-			onClose?.();
+			if (showInformationMessage) {
+				onClose?.();
+			} else {
+				onSetShowPrevConversations(false);
+				onSetShowHomeScreen(true);
+			}
```
Comment (kwinto) [inline src/webchat-ui/components/WebchatUI.tsx:789]: Close button should always close the widget. To get back to home screen, there is a back button.

There might be a case where it's not always true (e.g. some pop-up), here though it is not the case.

File: src/webchat-ui/components/WebchatUI.tsx [hunk 2]
```diff
@@ -768,14 +799,16 @@ export class WebchatUI extends React.PureComponent<
 			<>
 				<Header
 					onClose={handleOnClose}
-					onGoBack={handleOnGoBack}
+					onGoBack={!showInformationMessage ? handleOnGoBack : undefined}
```
Comment (kwinto) [inline src/webchat-ui/components/WebchatUI.tsx:802]: It's better for readability to not have an inversion if possible:
```javascript
onGoBack={showInformationMessage ? undefined : handleOnGoBack}
```

**Review by kwinto — CHANGES_REQUESTED** [review]
some smaller ones, otherwise looks good to me

File: src/webchat-ui/components/WebchatUI.tsx (hunk 2 above)
Comment (fabclj) [reply src/webchat-ui/components/WebchatUI.tsx]: fixed

File: src/webchat-ui/components/WebchatUI.tsx (hunk 1 above)
Comment (fabclj) [reply src/webchat-ui/components/WebchatUI.tsx]: fixed

**Review by kwinto — APPROVED** [review]

**Pull request merged**